pytorch_lightning/trainer/connectors/logger_connector.py

```diff
@@ -87,7 +87,7 @@ def log_metrics(self, metrics, grad_norm_dic, step=None):
         elif step is None:
             # added metrics by Lightning for convenience
             scalar_metrics['epoch'] = self.trainer.current_epoch
-            step = step if step is not None else self.trainer.global_step
+            step = self.trainer.global_step
```

Comment:
I believe #4132 tried to solve the problem of logging epoch on the x-axis in case of `*_epoch_end`, not sure why it was closed. I suggest by default it should set `global_step` for step_logs and `current_epoch` for epoch_logs and in case if `step` is provided in the metrics by the user, it should be replaced with that value.

Reply:
I agree. Even better if for epoch_logs the default is the current epoch.

Reply:
From feedback it is clear that the default should be `global_step`.

Reply:
please note that the log_metrics method here is recursive. 
At the bottom, it is calling itself with step (before) but now step is None. This means in the recursive call, step will be None and then it will add the epoch key. That is why you see the epoch key in the test as extra. Just wanted to point that out in case this was not clear.